data/block/blockV2.go

```diff
@@ -640,9 +640,7 @@ func (hv2 *HeaderV2) CheckFieldsForNil() error {
 	if err != nil {
 		return err
 	}
-	if hv2.ScheduledRootHash == nil {
-		return fmt.Errorf("%w in HeaderV2.ScheduledRootHash", data.ErrNilValue)
```

Comment:
Is there a situation in HeaderV2 where ScheduledRootHash is nil? It is set on CreateNewHeader, so also on the proposed headers, with the value of the current rootHash.

Reply:
Was removed for sovereign compatibility, as @sasurobert mentioned